### PR TITLE
Added UI Tests for SignIn Using Espresso

### PR DESCRIPTION
**Pull request by ishchow**
Added UI tests for signin using Espresso.

Signup UI tests were left out since that requires UI in the app to delete an account in order for the tests to be repeatable. That can come in a separate PR. However, the danger with having the user being able to delete the account is that they normally shouldn't be able to do that since that could really mess up with the database since a lot of stuff is tied in with their userID.

### CODE REVIEW

**Comment by ishchow** [comment]
Related #87

File: app/src/androidTest/java/com/example/atheneum/activities/FirebaseUIAuthActivitySignInTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        ViewInteraction appCompatImageButton = onView(
+                allOf(withContentDescription("Open navigation drawer"),
+                        childAtPosition(
+                                allOf(withId(R.id.toolbar),
+                                        childAtPosition(
+                                                withClassName(is("android.support.design.widget.AppBarLayout")),
+                                                0)),
+                                1),
+                        isDisplayed()));
+        appCompatImageButton.perform(click());
```
Comment (AY-Chen) [inline app/src/androidTest/java/com/example/atheneum/activities/FirebaseUIAuthActivitySignInTest.java]: I'm getting it failing here with 
```
android.support.test.espresso.NoMatchingViewException: No views in hierarchy found matching: (with content description: is "Open navigation drawer" and Child at position 1 in parent (with id: 2131231032 and Child at position 0 in parent with class name: is "android.support.design.widget.AppBarLayout") and is displayed on the screen to the user)
```

File: app/src/main/java/com/example/atheneum/activities/FirebaseUIAuthActivity.java [hunk 2]
```diff
@@ -115,7 +115,7 @@ private void generateSignInPage() {
                 AuthUI.getInstance()
                         .createSignInIntentBuilder()
                         .setAvailableProviders(providers)
-//                                        .setIsSmartLockEnabled(false) // Disabled for testing, used for password-less login
+                        .setIsSmartLockEnabled(false) // Used for password-less login, disabled to make UI tests easy
```
Comment (AY-Chen) [inline app/src/main/java/com/example/atheneum/activities/FirebaseUIAuthActivity.java]: This should be fine, but has problems if the user was logged in before. In my case it was still logged in when the app opened at the test failed immediately. 
For anyone running this just remember to log out.

File: app/src/androidTest/java/com/example/atheneum/activities/FirebaseUIAuthActivitySignInTest.java (hunk 1 above)
Comment (ishchow) [reply app/src/androidTest/java/com/example/atheneum/activities/FirebaseUIAuthActivitySignInTest.java]: Can someone else run the test to check if this is just a one off or not?

File: app/src/main/java/com/example/atheneum/activities/FirebaseUIAuthActivity.java (hunk 2 above)
Comment (ishchow) [reply app/src/main/java/com/example/atheneum/activities/FirebaseUIAuthActivity.java]: I made a new change that checks the user's auth status and their connectivity status. The test throws an assert if either the user is logged or they are offline. This is to ensure that the tests start in a reliable state.

**Comment by MarcusBoay** [comment]
I ran the test and it ran to completion.

**Comment by ishchow** [comment]
How to add new Espresso Tests using the Test Recorder:

1. Manually login to the app using the test user credentials (see the SignInTest class for the user/password).
2. Temporarily edit AndroidManifest.xml file with these changes. Don't commit these changes to git.
Edit the entry for the FirebaseAuthUIActivity and the MainActivity like this:
```
        <activity android:name=".activities.MainActivity">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />

                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
        <activity android:name=".activities.FirebaseUIAuthActivity">
            <!--<intent-filter>-->
                <!--<action android:name="android.intent.action.MAIN" />-->

                <!--<category android:name="android.intent.category.LAUNCHER" />-->
            <!--</intent-filter>-->
        </activity>
```
3. Now you can start using the Espresso Test Recorder, you'll start from the MainActivity instead of the Auth activity. This will make writing the test classes more convenient. Write your test class and test it in isolation from login/logout. DO NOT LOGOUT of the app during this stage or in any of your tests!
4. Reverse the changes to step 2. 
5. Add your test class to the AtheneumTestSuite in between SignIn.class and Signout.class! ENSURE THAT THE TEST CLASS GOES IN THE RIGHT PLACE!!!!! Or else the tests will fail due to improper signin/signout.
6. Hopefully at this point all the tests pass and now you can commit your changes!

**Pull request merged**